File: VB/WindowsFormsApplication31/RepositoryItemCustomToggleSwitch.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace WindowsFormsApplication31
    <UserRepositoryItem("RegisterCustomToggleSwitch")> _
    Public Class RepositoryItemCustomToggleSwitch
        Inherits RepositoryItemToggleSwitch

        Shared Sub New()
            RegisterCustomToggleSwitch()
        End Sub

        Public Const CustomEditName As String = "CustomToggleSwitch"

        Public Sub New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return CustomEditName
            End Get
        End Property

        Public Shared Sub RegisterCustomToggleSwitch()
            Dim img As Image = Nothing
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(CustomEditName, GetType(CustomToggleSwitch), GetType(RepositoryItemCustomToggleSwitch), GetType(CustomToggleSwitchViewInfo), New CustomToggleSwitchPainter(), True, img))
        End Sub

        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            BeginUpdate()
            Try
                MyBase.Assign(item)
                Dim source As RepositoryItemCustomToggleSwitch = TryCast(item, RepositoryItemCustomToggleSwitch)
                If source Is Nothing Then
                    Return
                End If
                '
            Finally
                EndUpdate()
            End Try
        End Sub
    End Class

End Namespace

File: CS/WindowsFormsApplication31/RepositoryItemCustomToggleSwitch.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace WindowsFormsApplication31 {
    [UserRepositoryItem("RegisterCustomToggleSwitch")]
    public class RepositoryItemCustomToggleSwitch : RepositoryItemToggleSwitch {
        static RepositoryItemCustomToggleSwitch() {
            RegisterCustomToggleSwitch();
        }

        public const string CustomEditName = "CustomToggleSwitch";

        public RepositoryItemCustomToggleSwitch() {
        }

        public override string EditorTypeName {
            get {
                return CustomEditName;
            }
        }

        public static void RegisterCustomToggleSwitch() {
            Image img = null;
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(CustomEditName, typeof(CustomToggleSwitch), typeof(RepositoryItemCustomToggleSwitch), typeof(CustomToggleSwitchViewInfo), new CustomToggleSwitchPainter(), true, img));
        }

        public override void Assign(RepositoryItem item) {
            BeginUpdate();
            try {
                base.Assign(item);
                RepositoryItemCustomToggleSwitch source = item as RepositoryItemCustomToggleSwitch;
                if(source == null)
                    return;
                //
            } finally {
                EndUpdate();
            }
        }
    }

}
